### PR TITLE
New packages: vala-panel, vala-panel-appmenu

### PR DESCRIPTION
**Pull request by purpurrot**

### CODE REVIEW

File: srcpkgs/cmake-vala/template [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+version=1
+revision=1
+wrksrc="cmake-vala-r${version}"
+build_style=cmake
+depends="cmake"
+short_desc="Vala CMake modules"
+maintainer="Juan RP <xtraeme@voidlinux.org>"
+license="BSD-3-Clause"
+homepage="https://gitlab.com/vala-panel-project/cmake-vala"
+distfiles="${homepage}/-/archive/r${version}/cmake-vala-r${version}.tar.gz"
```
Comment (maxice8) [inline srcpkgs/cmake-vala/template]: write out the homepage

File: srcpkgs/vala-panel-appmenu/template [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+configure_args="-DENABLE_XFCE=ON -DENABLE_VALAPANEL=ON
+ -DENABLE_MATE=ON -DENABLE_JAYATANA=ON -DENABLE_APPMENU_GTK_MODULE=ON
+ -DJAVA_INCLUDE_PATH=/usr/lib/jvm/openjdk/include
+ -DJAVA_JVM_LIBRARY=/usr/lib/jvm/openjdk/lib
+ -DJAVA_AWT_LIBRARY=/usr/lib/jvm/openjdk/lib/amd64"
+hostmakedepends="pkg-config"
+makedepends="vala-devel bamf-devel gtk+-devel gtk+3-devel glib-devel
+ libxkbcommon-devel openjdk libpeas-devel libdbusmenu-glib-devel cmake-vala
+ xfce4-panel-devel xfconf-devel libmate-panel-devel libXt-devel"
+depends="glib bamf "
```
Comment (maxice8) [inline srcpkgs/vala-panel-appmenu/template]: extra whitespace

File: srcpkgs/vala-panel-appmenu/template [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	short_desc="Mate Panel applet for appmenu module"
+	depends+="mate-panel"
+	pkg_install() {
+		vmove usr/lib/mate-panel
+		vmove usr/share/mate-panel
+	}
+}
+
+vala-panel-appmenu_package() {
+	depends+=" vala-panel"
```
Comment (maxice8) [inline srcpkgs/vala-panel-appmenu/template]: what ?

File: srcpkgs/vala-panel/template [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Template file for 'vala-panel'
+pkgname=vala-panel
+version=0.4.87
+revision=1
+build_style=cmake
+hostmakedepends="pkg-config"
```
Comment (maxice8) [inline srcpkgs/vala-panel/template]: pkgconf

File: srcpkgs/vala-panel/template [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+makedepends="vala-devel cmake-vala gtk+3-devel glib-devel"
```
Comment (maxice8) [inline srcpkgs/vala-panel/template]: libglib-devel

File: srcpkgs/vala-panel/template [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+depends="glib gtk+3 bamf"
```
Comment (maxice8) [inline srcpkgs/vala-panel/template]: won't this be found automatically by shlib tracking ?

File: srcpkgs/vala-panel/template [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+short_desc="Vala Panel is GTK+ 3 desktop panel inspired by LXPanel"
+maintainer="Juan RP <xtraeme@voidlinux.org>"
```
Comment (maxice8) [inline srcpkgs/vala-panel/template]: use your name

File: srcpkgs/vala-panel/template [hunk 8]
```diff
@@ -0,0 +1,22 @@
+# Template file for 'vala-panel'
+pkgname=vala-panel
+version=0.4.87
+revision=1
+build_style=cmake
+hostmakedepends="pkg-config"
+makedepends="vala-devel cmake-vala gtk+3-devel glib-devel"
+depends="glib gtk+3 bamf"
+short_desc="Vala Panel is GTK+ 3 desktop panel inspired by LXPanel"
```
Comment (maxice8) [inline srcpkgs/vala-panel/template]: GTK+3 desktop panel inspired by LXPanel

File: srcpkgs/vala-panel-appmenu/template [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+	depends+="xfce4-panel"
+	pkg_install() {
+		vmove usr/lib/xfce4
+		vmove usr/share/xfce4
+	}
+}
+
+mate-panel-appmenu_package() {
+	short_desc="Mate Panel applet for appmenu module"
+	depends+="mate-panel"
```
Comment (maxice8) [inline srcpkgs/vala-panel-appmenu/template]: depends="mate-panel"

File: srcpkgs/vala-panel-appmenu/template [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+	depends="glib"
+	pkg_install() {
+		vmove usr/libexec/vala-panel
+		vmove usr/share/dbus-1
+	}
+}
+
+xfce4-panel-appmenu_package() {
+	short_desc="XFCE Panel applet for appmenu module"
+	depends+="xfce4-panel"
```
Comment (maxice8) [inline srcpkgs/vala-panel-appmenu/template]: depends="xfce4-panel"

File: srcpkgs/vala-panel-appmenu/template [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+	pkg_install() {
+		vmove usr/lib/pkgconfig/appmenu-gtk3-parser.pc
+		vmove usr/lib/gtk-3.0
+		vmove usr/lib/libappmenu-gtk3*
+	}
+}
+
+appmenu-jayatana_package() {
+	short_desc="Global Menu for Java Swing applications"
+	depends+="openjdk libxkbcommon libdbusmenu-glib"
```
Comment (maxice8) [inline srcpkgs/vala-panel-appmenu/template]: depends="openjdk libxkbcommon libdbusmenu-glib"

File: srcpkgs/vala-panel-appmenu/template [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+	pkg_install() {
+		vmove usr/lib/pkgconfig/appmenu-gtk2-parser.pc
+		vmove usr/lib/gtk-2.0
+		vmove usr/lib/libappmenu-gtk2*
+	}
+}
+
+appmenu-gtk3-module_package() {
+	short_desc="GTK3 module for exposing menus"
+	depends+="gtk+3"
```
Comment (maxice8) [inline srcpkgs/vala-panel-appmenu/template]: will problably be found automatically

File: srcpkgs/vala-panel-appmenu/template [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+	short_desc="GTK module for exposing menus - development files"
+	depends+="appmenu-gtk3-module"
+	pkg_install() {
+		vmove usr/include
+	}
+}
+
+appmenu-gtk-module_package() {
+	short_desc="GTK module for exposing menus"
+	depends+="gtk+"
```
Comment (maxice8) [inline srcpkgs/vala-panel-appmenu/template]: will problably by found automatically

File: srcpkgs/vala-panel-appmenu/template [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+license="LGPL-3.0-or-later"
+homepage="https://gitlab.com/vala-panel-project/vala-panel-appmenu"
+distfiles="${homepage}/uploads/c0f85b42a2a228ad8c3413357e02fb28/vala-panel-appmenu-${version}.tar.xz"
+checksum=54a1b4f6ceca63eb3b79facc4327424bdd82f0fe3b7c6139326dbe0ec0824dcc
+subpackages="appmenu-gtk-module-devel appmenu-gtk-module appmenu-gtk3-module
+ appmenu-jayatana appmenu-registrar xfce4-panel-appmenu mate-panel-appmenu"
+
+appmenu-gtk-module-devel_package() {
+	short_desc="GTK module for exposing menus - development files"
+	depends+="appmenu-gtk3-module"
```
Comment (maxice8) [inline srcpkgs/vala-panel-appmenu/template]: depends=""

File: srcpkgs/vala-panel-appmenu/template [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+	pkg_install() {
```
Comment (maxice8) [inline srcpkgs/vala-panel-appmenu/template]: missing pkg-config files and .solinks

File: srcpkgs/vala-panel-appmenu/template [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+ libxkbcommon-devel openjdk libpeas-devel libdbusmenu-glib-devel cmake-vala
+ xfce4-panel-devel xfconf-devel libmate-panel-devel libXt-devel"
+depends="glib bamf "
+short_desc="Global Menu for Vala Panel"
+maintainer="Juan RP <xtraeme@voidlinux.org>"
+license="LGPL-3.0-or-later"
+homepage="https://gitlab.com/vala-panel-project/vala-panel-appmenu"
+distfiles="${homepage}/uploads/c0f85b42a2a228ad8c3413357e02fb28/vala-panel-appmenu-${version}.tar.xz"
+checksum=54a1b4f6ceca63eb3b79facc4327424bdd82f0fe3b7c6139326dbe0ec0824dcc
+subpackages="appmenu-gtk-module-devel appmenu-gtk-module appmenu-gtk3-module
```
Comment (maxice8) [inline srcpkgs/vala-panel-appmenu/template]: don't need to specificy

File: srcpkgs/vala-panel-appmenu/template (hunk 2 above)
Comment (maxice8) [reply srcpkgs/vala-panel-appmenu/template]: also will problably be found automatically

File: srcpkgs/cmake-vala/template [hunk 17]
```diff
@@ -0,0 +1,17 @@
+# Template file for 'cmake-vala'
+pkgname=cmake-vala
+version=1
+revision=1
```
Comment (maxice8) [inline srcpkgs/cmake-vala/template]: this will most likely be archs=noarch

File: srcpkgs/vala-panel-appmenu/template [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+pkgname=vala-panel-appmenu
+version=0.7.1
+revision=1
+build_style=cmake
+configure_args="-DENABLE_XFCE=ON -DENABLE_VALAPANEL=ON
+ -DENABLE_MATE=ON -DENABLE_JAYATANA=ON -DENABLE_APPMENU_GTK_MODULE=ON
+ -DJAVA_INCLUDE_PATH=/usr/lib/jvm/openjdk/include
+ -DJAVA_JVM_LIBRARY=/usr/lib/jvm/openjdk/lib
+ -DJAVA_AWT_LIBRARY=/usr/lib/jvm/openjdk/lib/amd64"
+hostmakedepends="pkg-config"
```
Comment (maxice8) [inline srcpkgs/vala-panel-appmenu/template]: pkgconf

File: srcpkgs/vala-panel-appmenu/template [hunk 19]
```diff
@@ -0,0 +1,91 @@
...
+makedepends="vala-devel bamf-devel gtk+-devel gtk+3-devel glib-devel
```
Comment (maxice8) [inline srcpkgs/vala-panel-appmenu/template]: libglib-devel

File: srcpkgs/vala-panel-appmenu/template (hunk 3 above)
Comment (purpurrot) [reply srcpkgs/vala-panel-appmenu/template]: vala-panel-appmenu needs libvalapanel from vala-panel

Comment (maxice8) [reply srcpkgs/vala-panel-appmenu/template]: you are defining vala-panel-appmenu as a subpackage while it is also the main package.

File: srcpkgs/cmake-vala/template (hunk 1 above)
Comment (purpurrot) [reply srcpkgs/cmake-vala/template]: fixed

File: srcpkgs/vala-panel-appmenu/template (hunk 2 above)
Comment (purpurrot) [reply srcpkgs/vala-panel-appmenu/template]: should i remove them then?

File: srcpkgs/vala-panel-appmenu/template (hunk 3 above)
Comment (purpurrot) [reply srcpkgs/vala-panel-appmenu/template]: ups i will correct that

File: srcpkgs/vala-panel/template (hunk 5 above)
Comment (purpurrot) [reply srcpkgs/vala-panel/template]: > This was done very poorly with no understanding for core xbps-src concepts like shlib tracking.
> 
> Please apply proposed changes.

sorry I am pretty new to void linux, but I will try to fix the templates

File: srcpkgs/vala-panel/template (hunk 7 above)
Comment (purpurrot) [reply srcpkgs/vala-panel/template]: ok

File: srcpkgs/vala-panel/template (hunk 8 above)
Comment (purpurrot) [reply srcpkgs/vala-panel/template]: fixed

File: srcpkgs/vala-panel-appmenu/template (hunk 9 above)
Comment (purpurrot) [reply srcpkgs/vala-panel-appmenu/template]: fixed

File: srcpkgs/vala-panel-appmenu/template (hunk 10 above)
Comment (purpurrot) [reply srcpkgs/vala-panel-appmenu/template]: fixed

File: srcpkgs/vala-panel-appmenu/template (hunk 11 above)
Comment (purpurrot) [reply srcpkgs/vala-panel-appmenu/template]: fixed

File: srcpkgs/vala-panel-appmenu/template (hunk 12 above)
Comment (purpurrot) [reply srcpkgs/vala-panel-appmenu/template]: ok

File: srcpkgs/vala-panel-appmenu/template (hunk 13 above)
Comment (purpurrot) [reply srcpkgs/vala-panel-appmenu/template]: fixed

File: srcpkgs/vala-panel-appmenu/template (hunk 14 above)
Comment (purpurrot) [reply srcpkgs/vala-panel-appmenu/template]: fixed

File: srcpkgs/vala-panel-appmenu/template (hunk 16 above)
Comment (purpurrot) [reply srcpkgs/vala-panel-appmenu/template]: ok

File: srcpkgs/cmake-vala/template (hunk 17 above)
Comment (purpurrot) [reply srcpkgs/cmake-vala/template]: ok, has been changed

File: srcpkgs/vala-panel-appmenu/template (hunk 19 above)
Comment (purpurrot) [reply srcpkgs/vala-panel-appmenu/template]: fixed

File: srcpkgs/vala-panel-appmenu/template (hunk 18 above)
Comment (purpurrot) [reply srcpkgs/vala-panel-appmenu/template]: fixed

File: srcpkgs/vala-panel/template (hunk 4 above)
Comment (purpurrot) [reply srcpkgs/vala-panel/template]: fixed

File: srcpkgs/vala-panel-appmenu/template (hunk 15 above)
Comment (maxice8) [reply srcpkgs/vala-panel-appmenu/template]: This is the only -devel package so it should hold all pkg-config files.

File: srcpkgs/vala-panel-appmenu/template [hunk 20]
```diff
@@ -0,0 +1,79 @@
...
+license="LGPL-3.0-or-later"
+homepage="https://gitlab.com/vala-panel-project/vala-panel-appmenu"
+distfiles="https://gitlab.com/vala-panel-project/vala-panel-appmenu/uploads/c0f85b42a2a228ad8c3413357e02fb28/vala-panel-appmenu-${version}.tar.xz"
+checksum=54a1b4f6ceca63eb3b79facc4327424bdd82f0fe3b7c6139326dbe0ec0824dcc
+
+appmenu-gtk-module-devel_package() {
+	short_desc="GTK module for exposing menus - development files"
+	pkg_install() {
+		vmove usr/include
+		vmove usr/lib/pkgconfig/appmenu-gtk*.pc
```
Comment (maxice8) [inline srcpkgs/vala-panel-appmenu/template]: you can do usr/lib/pkgconfig

File: srcpkgs/vala-panel/template [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+short_desc="GTK+3 desktop panel inspired by LXPanel"
+maintainer="John Eira <john.eira@web.de>"
+license="LGPL-3.0-or-later"
+homepage="https://gitlab.com/vala-panel-project/vala-panel"
+distfiles="https://gitlab.com/vala-panel-project/vala-panel/uploads/8b42536f91a1d6a185350ce2e3b7e91f/vala-panel-${version}.tar.xz"
+checksum=625c12672553e1807f98f58c43c631c081938652756daddf0f6e37cffe392d08
+
+vala-panel-devel_package() {
+	short_desc+=" - development files"
+	depends="vala-panel"
```
Comment (maxice8) [inline srcpkgs/vala-panel/template]: please restrict the version like it is done in other -devel packages

File: srcpkgs/vala-panel-appmenu/template [hunk 22]
```diff
@@ -0,0 +1,79 @@
...
+ libxkbcommon-devel openjdk libpeas-devel libdbusmenu-glib-devel cmake-vala
+ xfce4-panel-devel xfconf-devel libmate-panel-devel libXt-devel"
+short_desc="Global Menu for Vala Panel"
+maintainer="John Eira <john.eira@web.de>"
+license="LGPL-3.0-or-later"
+homepage="https://gitlab.com/vala-panel-project/vala-panel-appmenu"
+distfiles="https://gitlab.com/vala-panel-project/vala-panel-appmenu/uploads/c0f85b42a2a228ad8c3413357e02fb28/vala-panel-appmenu-${version}.tar.xz"
+checksum=54a1b4f6ceca63eb3b79facc4327424bdd82f0fe3b7c6139326dbe0ec0824dcc
+
+appmenu-gtk-module-devel_package() {
```
Comment (maxice8) [inline srcpkgs/vala-panel-appmenu/template]: it should depend on all other packages and in the main package. use version restriction.

File: srcpkgs/cmake-vala/template [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+wrksrc="cmake-vala-r${version}"
+build_style=cmake
+depends="cmake"
+short_desc="Vala CMake modules"
+maintainer="John Eira <john.eira@web.de>"
+license="BSD-3-Clause"
+homepage="https://gitlab.com/vala-panel-project/cmake-vala"
+distfiles="https://gitlab.com/vala-panel-project/cmake-vala/-/archive/r${version}/cmake-vala-r${version}.tar.gz"
+checksum=f8801ce01ed7ced690229ef81c0809af212166d585a0de8cf2b276444fd9f942
+archs=noarch
```
Comment (maxice8) [inline srcpkgs/cmake-vala/template]: should be between build_style= and wkrsrc=

File: srcpkgs/vala-panel/template (hunk 21 above)
Comment (purpurrot) [reply srcpkgs/vala-panel/template]: ok

File: srcpkgs/vala-panel-appmenu/template (hunk 20 above)
Comment (purpurrot) [reply srcpkgs/vala-panel-appmenu/template]: ok

File: srcpkgs/cmake-vala/template (hunk 23 above)
Comment (purpurrot) [reply srcpkgs/cmake-vala/template]: ok

File: srcpkgs/vala-panel-appmenu/template (hunk 22 above)
Comment (purpurrot) [reply srcpkgs/vala-panel-appmenu/template]: fixed

**Comment by purpurrot** [comment]
Is there a difference between libglib-devel and glib-devel?
vala-panel seems to need glib-devel or otherwise the build fails.

**Comment by maxice8** [comment]
On Sat, Feb 23, 2019 at 10:12:39AM -0800, purpurrot wrote:
> Is there a difference between libglib-devel and glib-devel?
> vala-panel seems to need glib-devel or otherwise the build fails.
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/9110#issuecomment-466674877

libglib-devel holds files that you use when compiling and linking against libglib

libglib goes into makedepends and so does libglib-devel

glib-devel is tooling that you need to run on the host system to generate code and
other things related to libglib.

Since it is tooling that needs to be executed it goes into hostmakedepends=

If you are having errors with stuff like glib-mkenums then you need to add
glib-devel to hostmakedepends=

**Comment by purpurrot** [comment]
Thanks for the explanation!
I added glib-devels to the hostmakedepends and it builds on my machine.

**Pull request closed without merging**